File: Dijkstra.py
INF = int(1e9)

def Dijkstra(s):
	d[s] = 0
	for i in range(n):
		mn, mni = INF, -1
		for j in range(n):
			if d[j] < mn and not u[j]:
				mn, mni = d[j], j
		if mn == INF:
			break
		u[mni] = 1
		for to, r in g[mni]:
			if d[to] > d[mni] + r:
				d[to] = d[mni] + r
			

n, m = map(int, input().split())
g = [[] for i in range(n)]
d = [INF] * n
u = [0] * n

for i in range(m):
	a, b, r = map(int, input().split())
	a -= 1
	b -= 1
	g[a].append((b, r))
	g[b].append((a, r))
s, f = map(int, input().split())
Dijkstra(s - 1)
if d[f - 1] == INF:
	print(-1)
else:
	print(d[f - 1])

File: Dijkstra.cpp
#include <iostream>
#include <vector>

using namespace std;

#define INF int(1e9)
#define fs first
#define sc second

int n, m;
vector <vector<pair<int, int>>> g;
vector<int> d;
vector<char> u;

void Dijkstra(int s)
{
	d[s] = 0;
	for (int i = 0; i < n; i++)
	{
		int mn = INF, mni = -1;
		for (int j = 0; j < n; j++)
			if (d[j] < mn && !u[j])
			{
				mn = d[j];
				mni = j;
			}
		if (mn == INF)
			break;
		u[mni] = 1;
		for (auto to : g[mni])
			if (d[to.fs] > d[mni] + to.sc)
				d[to.fs] = d[mni] + to.sc;
	}
}

int main()
{
	cin >> n >> m;
	g.resize(n);
	d.resize(n, INF);
	u.resize(n, 0);
	for (int i = 0; i < m; i++)
	{
		int a, b, r;
		cin >> a >> b >> r;
		a--; b--;
		g[a].push_back({ b, r });
		g[b].push_back({ a, r });
	}
	int s, f;
	cin >> s >> f;
	Dijkstra(s - 1);
	if (d[f - 1] == INF)
		cout << -1;
	else
		cout << d[f - 1];
}
